Task 2

shruthiqualitest opened the pull request:
Tests for Task-2

shruthiqualitest committed:
added test for task-2

shruthiqualitest committed:
added test for task-2

shruthiqualitest committed:
added test for task-2

dStone7 commented on src/main/java/com/qualitestgroup/java/assessment/framework/pages/DemoPage.java:
```diff
@@ -24,6 +36,30 @@ public DemoPage(WebDriver driver) {
+        if (value[3].equals("check")) {
+
+            isStep = true;
+        }
```
Why not just return false if it does not equal "check"? It would be wasted execution time to continue after a fail.

dStone7 reviewed (changes requested):
Please post an image of the test passing.
Review all feedback and make any necessary changes.
The Pull Request also needs a better description of the changes that were implemented.

shruthiqualitest committed:
Create Utility methods

create utility methods and update all tests.

shruthiqualitest committed:
Create Utility methods

create utility methods and update all tests.

shruthiqualitest committed:
updates base test plans

shruthiqualitest committed:
Merge branch 'Task-2' of https://github.com/shruthiqualitest/javacodingassessment into Task-2

shruthiqualitest committed:
updated tests

Updated all tests based on code review comments. Made use of iid's to find elements.

shruthiqualitest commented:
![image](https://user-images.githubusercontent.com/97317255/150452624-bf14ff80-2b86-4eb2-aa29-7d4e6b40eb0e.png)

shruthiqualitest commented on src/main/java/com/qualitestgroup/java/assessment/framework/pages/DemoPage.java:
```diff
@@ -16,6 +16,18 @@
+    private WebElement expandbutton;
+
+    String angularChckBx = "(//li[contains(@class,'rct-node-expanded')])[3]//li[2]//span[@class='rct-checkbox']";
+    String svcElement = "/*[name()='svg']";
```
updated

shruthiqualitest commented on src/main/java/com/qualitestgroup/java/assessment/framework/pages/DemoPage.java:
```diff
@@ -16,6 +16,18 @@
+    @FindBy(xpath = "//div[@class='category-cards']/div")
+    private WebElement elementCard;
+
+    @FindBy(xpath = "(//ul[@class='menu-list']/li[2])")
```
At the moment, there aren't other ways for us to fetch the elements apart from using xpath. There are no id's and using text is as fragile as xpath.

dStone7 replied:
I dont agree. This xpath can be improved. It is too unstable to have an xpath return multiple elements when we still have additional ways to narrow the search. This would be a better xpath: "//ul[@class='menu-list']/li/span[text()='Check Box']"

shruthiqualitest commented on src/main/java/com/qualitestgroup/java/assessment/framework/pages/DemoPage.java:
```diff
@@ -16,6 +16,18 @@
     public CommonPage page;
     public GlobalVars global;
 
+    @FindBy(xpath = "//div[@class='category-cards']/div")
```
At the moment, there aren't other ways for us to fetch the elements apart from using xpath. There are no id's and using text is as fragile as xpath.

dStone7 replied:
I understand you point of view, but in my experience a page element is less likely to have its text change (unless it is dynamic) versus the page structure changing in some minor way due to other factors. Plus it is easier to maintain in case of future changes. In my opinion this would be a better xpath: "//div[@class='card-body']/h5[text()='Elements']"

dStone7 commented:
It looks like you included all the framework updates on this branch. It would have been better to create a separate branch for those updates since these changes are not related to Task 2.

shruthiqualitest committed:
updated to use css

updated to use css for finding web elements.
